vendor/github.com/urso/ucfg/error.go

```diff
@@ -0,0 +1,261 @@
+package ucfg
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"runtime/debug"
+)
+
+type Error interface {
+	error
+	Reason() error
+
+	// error class, one of ErrConfig, ErrImplementation, ErrUnknown
+	Class() error
+
+	Message() string
+
+	// [optional] path of config element error occured for
+	Path() string
+
+	// [optional] stack trace
+	Trace() string
+}
+
+type baseError struct {
+	reason  error
+	class   error
+	message string
+}
+
+type criticalError struct {
+	baseError
+	trace string
+}
+
+type pathError struct {
+	baseError
+	meta *Meta
+	path string
+}
+
+// error Reasons
+var (
+	ErrMissing = errors.New("missing field")
+
+	ErrTypeNoArray = errors.New("field is no array")
+
+	ErrTypeMismatch = errors.New("type mismatch")
+
+	ErrKeyTypeNotString = errors.New("key must be a string")
+
+	ErrIndexOutOfRange = errors.New("out of range index")
+
+	ErrPointerRequired = errors.New("pointer required for unpacking configurations")
+
+	ErrArraySizeMistach = errors.New("Array size mismatch")
+
+	ErrExpectedObject = errors.New("expected object")
+
+	ErrNilConfig = errors.New("config is nil")
+
+	ErrNilValue = errors.New("nil value is invalid")
+
+	ErrTODO = errors.New("TODO - implement me")
```

Comment:
Looks like this is no longer used and can be removed in your next batch of ucfg changes.